### ISSUE TITLE
build a php backend

### ISSUE THREAD

**Issue by fernyb** [fernyb/Sequel-J#14]
For ease of installation build a php backend this will make it easy to install and would really compete with phpMyAdmin.
But lets keep the ruby server.
- Response from the server should all be JSON.

**Comment by fernyb** [comment]
I started with some code but anyone feel free to undo what I did with php.

**Comment by joehoyle** [comment]
I am a full time PHP developer (well, WordPress developer) so I would be happy to take this on

**Comment by fernyb** [comment]
Cool, I've added you are a collaborator to the project.
I think you should be able to push directly now.

**Comment by joehoyle** [comment]
Cool. I Have been thinking about the best way to do the PHP backend. Currently you put SlimPHP up, which looks to be a cool framework. The only downside is it requires mod_rewrite, and this could slightly increases the barrier to entry for server, IIS I believe can be a pain with mod_rewrite.

We could solve this by just having one PHP file, and pas everything via query params (now we use SJAPIRequest this will be much easier):

`api.php?endpoint=connect&username=root&password=root`

And so on. The downside to this is we would have to write different URLs for the Cappuccino app to sent to, depending on which server is being used. This isn't a problem, but it would require the app to be "aware" of which server to use.

I am open to suggestion here, on one hand I would like the PHP to be really easy to set up, and one PHP file is about as easy as it can get. However, I don't want to create unnecessary complexity for the potentially small amount of people who won't have `mod_rewrite` working.

My main motivation is phpMyAdmin does not use anything like that, and we want everyone who currently uses phpMyAdmin to be able to use Sequel-J.

**Comment by fernyb** [comment]
we should pass all endpoints to one file "api.php" with a parameter called "endpoint" which is the endpoint name.
We actually don't need front-end to know which server its using because I can have the ruby server catch api.php and have it route to appropriate route.

**Comment by joehoyle** [comment]
Ok sounds good

**Comment by joehoyle** [comment]
One thing which I have changed in the PHP backend is "ends of endpoints". Previously you were doing this: /table_headers/my_table_name

However, as we are now using the ?endpoint query param, it probably makes more sense to do it this way: ?endpoint=table_headers&table=my_table_name

And in 
Objj

``` Objective-J
[[SJAPIRequest sendRequestToEndpoint:@"table_headers" withOptions:{table: table_name} callback...
```

Just wanted to makes sure this was ok, as I expect you would have to modify the ruby backend accordingly.

**Comment by fernyb** [comment]
Yeah, the ruby backend has already been modified it looks for endpoint parameter and table parameter.

**Comment by joehoyle** [comment]
Ok, I just updated table_info and table_headers endpoints to do the same, so the ruby side might need updating

**Comment by guilhermeaiolfi** [comment]
If Ruby is not a requirement any more. Maybe it should be removed from the README page:

Requirements For Server

Ruby 1.9.2
RubyGems (Sinatra, Bundler)

I'm really looking forward to test Sequel-J.

**Comment by joehoyle** [comment]
@guilhermeaiolfi I believe the current plan is to keep both backens, at least for the time being.

**Comment by guilhermeaiolfi** [comment]
"keep both" meaning that if I'm going to use the php backend I still need to have ruby?

If that's the case and I remember correctly, some time ago, I saw some code that looked like the ruby backend would be doing the routing, even if the php backend was doing the actual work. Is that right?

If all info for actions is being passed as parameters(param1=...&param2=...) why don't set a global var to point to the right backend and use just one or another?

**Comment by joehoyle** [comment]
You can either use the PHP backend or ruby one, they are not used "together". Just set the path of either in the SJConstants file. If you use the PHP one, you won't have to run a separate web server if you are already running the app from one (whereas I believe you can't use the ruby servet to serve the app).
